src/Microsoft.ML.CodeGenerator/Templates/Console/Annotation.tt

```diff
@@ -0,0 +1,19 @@
+﻿<#@ output extension=".txt" #>
+<#+
+void CLI_Annotation()
+{#>
+//*****************************************************************************************
+//*                                                                                       *
+//* This is an auto-generated file by Microsoft ML.NET CLI (Command-Line Interface) tool. *
```

Comment:
⚠️ This is missing the `<auto-generated/>` comment, which means diagnostics will be reported in the file even though it is generated.

Reply:
My thinking is that these code files aren't true "auto generated" files. I expect users to update them. Especially considering that we are putting hard-coded paths in some of the files, and we are making `public` types, that I assume most users will want to make `internal`.

So what is the appropriate pattern here? I think it is similar to "scaffolding" in ASP.NET. Or even "Add -> new Item" in VS. Do they put any "this was generated" comments at the top there?

Reply:
For this case, the ideal header would be more like a template, with links explaining what the file is and where to find documentation of expected changes users would need to make. You would not need the `<auto-generated/>` comment for templates (one-time code generation).

Reply:
@maryamariyan - can you log an issue to address this? I think we should make that change separately and we will need @JakeRadMSFT @LittleLittleCloud and @rustd input as well.

Reply:
Created https://github.com/dotnet/machinelearning/issues/4387